ompi/mca/osc/ucx/osc_ucx_request.h

```diff
@@ -49,8 +54,10 @@ typedef struct ompi_osc_ucx_internal_request {
 #define OMPI_OSC_UCX_REQUEST_RETURN(req)                                \
     do {                                                                \
         OMPI_REQUEST_FINI(&request->super);                             \
+        OPAL_THREAD_LOCK(&mca_osc_ucx_component.lock);                  \
```

Comment:
This lock should be unnecessary. Open MPI's free list functions (opal_free_list_get, opal_free_list_return, etc) are conditionally thread safe already.

Reply:
deleted